src/System.Linq.Expressions/src/System/Linq/Expressions/TypeBinaryExpression.cs

```diff
@@ -60,7 +60,7 @@ internal Expression ReduceTypeEqual()
                     // either matches or is its type argument (T to its T?).
                     if (cType.GetNonNullableType() != TypeOperand.GetNonNullableType())
                     {
-                        return Expression.Block(Expression, Expression.Constant(value: false));
+                        return Expression.Block(Expression, Utils.Constant(value: false));
```

Comment:
TypeBinaryExpression is not reducible via `Reduce`; the reduced expression is only used by the compiler and interpreter and never leaks to user surface.